src/use-extra-deps/index.ts

```diff
@@ -0,0 +1,85 @@
+import isFunction from 'lodash/isFunction'
+import mapValues from 'lodash/mapValues'
+import omitBy from 'lodash/omitBy'
+import pickBy from 'lodash/pickBy'
+import values from 'lodash/values'
+import * as React from 'react'
+
+type $ObjMap<T extends {}, F extends (v: any) => any> = {
+  [K in keyof T]: F extends (v: T[K]) => infer R ? R : never
+}
+
+// Dependencies that are safe to use in the normal `useEffect` deps array
+export type PrimitiveDep = boolean | string | number | null | void | Symbol
+
+// Wrapper around a function that has been wrapped in `useSafeCallback`. This
+// type is here to avoid cyclical dependencies.
+export type CallbackFn<F> = F
```

Comment:
Adding a TS alias here instead of `opaque` as opaque is not available in TS. I think this is a good substitution for this case as we're basically saying `CallbackFn<F>` is really `F`